services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,20 +43,46 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
-
+  // Send an event to the thermal manager queue
+  if (xQueueSend(thermalMgrQueueHandle, event, 1000) !=  pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  // Send event to handle OS interrupt
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_HANDLE_OS_CMD;
+  if (xQueueSendFromISR(thermalMgrQueueHandle, &event, NULL) != pdTRUE) {
+    LOG_ERROR_CODE(ERR_CODE_QUEUE_FULL);
+  }
```

Comment:
should just be calling thermalMgrSendEvent from this to send the event. avoid logging directly from ISRs, do this in thermal mgr send event function

Reply:
It looks like the xQueueSend function is not meant to be called through an ISR, so I used this instead. Maybe I should use xQueueSendFromISR inside thermalMgrSendEvent and then call thermalMgrSendEvent in this ISR?

Reply:
Does this also mean that I shouldn't be logging anything in thermalMgrSendEvent since it would get called from the ISR?